Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean

```diff
@@ -94,22 +140,46 @@ lemma negPart_nonneg (a : A) :
     0 ≤ a⁻ :=
   cfcₙ_nonneg (fun x _ ↦ by positivity)
 
+lemma posPart_eq_of_eq_sub_negPart {a b : A} (hab : a = b - a⁻) (hb : 0 ≤ b := by cfc_tac) :
+    a⁺ = b := by
+  have ha := hab.symm ▸ hb.isSelfAdjoint.sub (negPart_nonneg a).isSelfAdjoint
+  nth_rw 1 [← posPart_sub_negPart a] at hab
+  simpa using hab
+
+lemma negPart_eq_of_eq_PosPart_sub {a c : A} (hac : a = a⁺ - c) (hc : 0 ≤ c := by cfc_tac) :
+    a⁻ = c := by
+  have ha := hac.symm ▸ (posPart_nonneg a).isSelfAdjoint.sub hc.isSelfAdjoint
+  nth_rw 1 [← posPart_sub_negPart a] at hac
+  simpa using hac
+
+lemma le_posPart {a : A} (ha : IsSelfAdjoint a) : a ≤ a⁺ := by
+  simpa [posPart_sub_negPart a] using sub_le_self a⁺ (negPart_nonneg a)
+
+lemma neg_negPart_le {a : A} (ha : IsSelfAdjoint a) : -a⁻ ≤ a := by
+  simpa only [posPart_sub_negPart a, ← sub_eq_add_neg]
+    using le_add_of_nonneg_left (a := -a⁻) (posPart_nonneg a)
+
 variable [NonnegSpectrumClass ℝ A]
 
-lemma eq_posPart_iff (a : A) : a = a⁺ ↔ 0 ≤ a := by
+lemma posPart_eq_self (a : A) : a⁺ = a ↔ 0 ≤ a := by
   refine ⟨fun ha ↦ ha ▸ posPart_nonneg a, fun ha ↦ ?_⟩
-  conv_lhs => rw [← cfcₙ_id ℝ a]
+  conv_rhs => rw [← cfcₙ_id ℝ a]
   rw [posPart_def]
   refine cfcₙ_congr (fun x hx ↦ ?_)
   simpa [_root_.posPart_def] using quasispectrum_nonneg_of_nonneg a ha x hx
 
-lemma negPart_eq_zero_iff (a : A) (ha : IsSelfAdjoint a) :
+@[deprecated posPart_eq_self (since := "2024-11-18")]
+lemma eq_posPart_iff (a : A) : a = a⁺ ↔ 0 ≤ a := by
+  rw [eq_comm, posPart_eq_self]
+
+lemma negPart_eq_zero_iff (a : A) (ha : IsSelfAdjoint a := by cfc_tac) :
     a⁻ = 0 ↔ 0 ≤ a := by
-  rw [← eq_posPart_iff]
+  rw [← posPart_eq_self, eq_comm (b := a)]
   nth_rw 2 [← posPart_sub_negPart a]
   simp
 
-lemma eq_negPart_iff (a : A) : a = -a⁻ ↔ a ≤ 0 := by
+lemma negPart_eq_neg (a : A) : a⁻ = -a ↔ a ≤ 0 := by
+  rw [← neg_inj, neg_neg, eq_comm]
```

Comment:
```suggestion
lemma negPart_eq_neg_iff (a : A) : a⁻ = -a ↔ a ≤ 0 := by
  rw [← neg_inj, neg_neg, eq_comm]
```
This feels a bit more in line with the rest, no?

Reply:
no, I updated these names to match [posPart_eq_self](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Group/PosPart.html#posPart_eq_self) and [negPart_eq_neg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Group/PosPart.html#negPart_eq_neg)